operator/src/main/java/org/bf2/operator/operands/Canary.java

```diff
@@ -343,13 +347,34 @@ private List<EnvVar> buildEnvVar(ManagedKafka managedKafka, Deployment current)
             addEnvVarFromSecret(envVars, "SASL_PASSWORD",SecuritySecretManager.canarySaslSecretName(managedKafka) , SecuritySecretManager.SASL_PASSWORD);
         }
         envVars.add(new EnvVarBuilder().withName("STATUS_TIME_WINDOW_MS").withValue(String.valueOf(statusTimeWindowMs)).build());
-        return envVars;
+        return applyEnvironmentOverride(envVars, this.overrideManager.getCanaryOverride(managedKafka.getSpec().getVersions().getStrimzi()).getEnv());
     }
 
     private List<ContainerPort> buildContainerPorts() {
         return Collections.singletonList(new ContainerPortBuilder().withName(METRICS_PORT_NAME).withContainerPort(METRICS_PORT).build());
     }
 
+    private List<EnvVar> applyEnvironmentOverride(List<EnvVar> originals, List<EnvVar> overrides) {
```

Comment:
This might be better placed in `OperandOverride` itself.

Reply:
Done